pkg/network/monitor/monitor.go

```diff
@@ -0,0 +1,176 @@
+package monitor
+
+import (
+	"sync"
+	"syscall"
+
+	"github.com/vishvananda/netlink"
+	"k8s.io/klog"
+)
+
+// netlink event type
+// link:  RTM_NEWLINK  RTM_DELLINK
+// addr:  RTM_NEWADDR  RTM_DELADDR
+// route: RTM_NEWROUTE RTM_DELROUTE
+
+type Monitor struct {
+	handlers map[int]Handler
+	mutex    sync.RWMutex
+
+	done chan struct{}
+
+	// every monitor can start/stop once
+	startOnce sync.Once
+	stopOnce  sync.Once
+}
+
+func NewMonitor() *Monitor {
+	return &Monitor{
+		handlers:  make(map[int]Handler),
+		mutex:     sync.RWMutex{},
+		done:      make(chan struct{}),
```

Comment:
Can we pass a context and use ctx.Done() as the done channel. No one is writing to or close it as for as I can see